internal/service/searchindex/resource_search_index.go

```diff
@@ -320,15 +319,39 @@ func resourceMongoDBAtlasSearchIndexRead(ctx context.Context, d *schema.Resource
 		return diag.Errorf("error setting `analyzer` for search index (%s): %s", d.Id(), err)
 	}
 
-	if analyzers := searchIndex.GetAnalyzers(); len(analyzers) > 0 {
+	if searchIndex.Type != nil && *searchIndex.Type == vectorSearch {
```

Comment:
why are we changing the order and logic of the if statements here?

Reply:
now we want to have empty arrays (instead of nil pointers) so for instance they're updated correctly.  but depending on the vector type some arrays can't not be sent even empty.    basically here we're doing the same logic as we do in Create.